File: python/cdk8s-crd/main.py
#!/usr/bin/env python3
from constructs import Construct
from cdk8s import App, Chart

from imports.io import cert_manager


class Certificate(Chart):
    def __init__(self, scope: Construct, id: str):
        super().__init__(scope, id)

        cert_manager.Certificate(
            self,
            "Certificate",
            spec=cert_manager.CertificateSpec(
                secret_name="secret",
                issuer_ref=cert_manager.CertificateSpecIssuerRef(name="issuer"),
            ),
        )


app = App()
Certificate(app, "dashboard")

app.synth()

File: go/cdk8s-crd/main.go
package main

import (
	"example.com/cdk8s-crd/imports/certmanagerio"
	"github.com/aws/constructs-go/constructs/v10"
	"github.com/aws/jsii-runtime-go"
	"github.com/cdk8s-team/cdk8s-core-go/cdk8s/v2"
)

func NewCertificate(scope constructs.Construct, id string) cdk8s.Chart {

	chart := cdk8s.NewChart(scope, jsii.String(id), nil)

	certmanagerio.NewCertificate(chart, jsii.String("Certificate"), &certmanagerio.CertificateProps{
		Spec: &certmanagerio.CertificateSpec{
			SecretName: jsii.String("secret"),
			IssuerRef: &certmanagerio.CertificateSpecIssuerRef{
				Name: jsii.String("issuer"),
			},
		},
	})

	return chart
}

func main() {
	app := cdk8s.NewApp(nil)
	NewCertificate(app, "certificate")
	app.Synth()
}
